apps/events-helsinki/src/domain/search/eventSearch/utils.tsx

```diff
@@ -366,3 +368,37 @@ export const getEventUrl = (
     locale
   );
 };
+
+export const getEventListLinkUrl = (
+  event: EventFields,
+  router: NextRouter,
+  locale: AppLanguage
+) => {
+  const search = router.asPath.split('?')[1];
+  return (
+    routerHelper.getLocalizedCmsItemUrl(
+      ROUTES.EVENTS,
+      { eventId: event.id },
+      locale
+    ) + (search ? `?${search}` : '')
+  );
+};
+
+export const getKeywordOnClickHandler =
+  (
+    router: NextRouter,
+    locale: AppLanguage,
+    type: 'dateType' | 'isFree' | 'text',
+    value = ''
+  ) =>
+  () => {
+    const search = getSearchQuery({
+      ...EVENT_DEFAULT_SEARCH_FILTERS,
+      dateTypes: type === 'dateType' ? [value] : [],
+      isFree: type === 'isFree',
+      text: type === 'text' ? [value] : [],
+    });
+
+    router.push(`${routerHelper.getI18nPath(ROUTES.SEARCH, locale)}${search}`);
```

Comment:
You omitted the getSearchQuery call from your suggestion.
Taking into account that `getSearchQuery` is
```typescript
export const getSearchQuery = (filters: Filters): string => {
  const newFilters: MappedFilters = {
    ...filters,
    end: formatDate(filters.end, 'yyyy-MM-dd'),
    isFree: filters.isFree ? true : undefined,
    start: formatDate(filters.start, 'yyyy-MM-dd'),
  };

  if (newFilters.end || newFilters.start) {
    delete newFilters.dateTypes;
  }

  if (newFilters.q?.length && !newFilters.q[0]) {
    delete newFilters.q;
  }

  return buildQueryFromObject(newFilters);
};
```
and that `buildQueryFromObject` is
```typescript
const buildQueryFromObject = (obj: { [key: string]: any }) => {
  const query: string[] = [];

  forEach(obj, (filter, key) => {
    if (!isEmpty(filter) || isNumber(filter) || typeof filter === 'boolean') {
      if (isArray(filter)) {
        const items: Array<string | number> = [];

        forEach(filter, (item: string | number) => {
          items.push(encodeURIComponent(item));
        });

        query.push(`${key}=${items.join(',')}`);
      } else if (!isNil(filter)) {
        query.push(`${key}=${encodeURIComponent(filter)}`);
      }
    }
  });

  return query.length ? `?${query.join('&')}` : '';
};
```
this will most probably change the implementation's functionality.
I won't change this now as taking into account all what could change
here is not straightforward but would take some time. I'll just go
with the current functionality which is how it was before this PR.